### PR TITLE
[Dev][AMD] Add AMD RDNA Arch

### PR DESCRIPTION
**Pull request by Cunxiao2002**

### CODE REVIEW

**Comment by LeiWang1999** [comment]
Hi @Cunxiao2002 , thanks for your contribution, would you mind add simple test case for this arch?

**Comment by Cunxiao2002** [comment]
@microsoft-github-policy-service agree

File: bitblas/builder/lib_generator/__init__.py [hunk 1]
```diff
@@ -30,24 +30,38 @@ def load_lib(self):
...
-            "'-fPIC'",
-            "-lineinfo",
-            "--shared",
-            src.name,
-            "-lcuda",
-            "-gencode",
-            f"arch=compute_{compute_version},code=sm_{compute_version}",
-        ]
+        platform = arch.platform
+        if (platform == "CUDA"):
```
Comment (LeiWang1999) [inline bitblas/builder/lib_generator/__init__.py:34]: I don't think we need brackets there.

**Comment by Cunxiao2002** [comment]
Thank you for your recommendations and I will improve my code based on them.

**Comment by LeiWang1999** [comment]
I've reviewed the code and left some comments.

**Comment by LeiWang1999** [comment]
@Cunxiao2002  Please fix the lint :)

**Comment by Cunxiao2002** [comment]
> @Cunxiao2002 Please fix the lint :)

I'll fix it right away, sorry.

**Review by LeiWang1999 — APPROVED** [review]

**Pull request merged**